lib/filewriter.js

```diff
@@ -138,16 +138,18 @@ FileWriter.prototype = {
             var guard, error = null;
             function finish(err) {
                 if (guard && global.clearTimeout) clearTimeout(guard);
+                var err2;
                 guard = null;
-                if (fd !== null) fs.close(fd, function(){});
+                if (fd !== null) try { fs.closeSync(fd) } catch (e) { err2 = e }
                 fd = null;
-                return error ? cb(error) : cb(err);
+                return cb(error || err || err2);
             }
             try {
                 guard = setTimeout(function() {
-                    if (fd !== null) fs.close(fd, function(){});
+                    var err;
+                    if (fd !== null) try { fs.closeSync(fd) } catch (e) { err = e }
                     error = new Error("timed out waiting for last write");
-                    return finish(error);
+                    return finish(error || err);
```

Comment:
correct; the usage was intentional to emphasize that two different errors can be in play